### DISCUSSION TITLE
if terminal width < than a size, put the user input on a new line

### DISCUSSION THREAD

**Discussion by mirsella** [JanDeDobbeleer/oh-my-posh#3148] [Troubleshoot] [answered]
hey, i don't know if it's already possible, sorry if it is i didn't find it.

in our themes.json something like a min_width value, and if the terminal is smaller the user input is put on a new line ?

![image](https://user-images.githubusercontent.com/45905567/204872392-6fd158f7-4446-44a9-a9b3-e8a87472a9fe.png)

thanks !

**Comment by JanDeDobbeleer** [comment, marked as answer]
Hmm. No that we don't have, however in this case the rprompt shouldn't print. I'll check if that still works correctly.

**Reply by mirsella** [reply]
I can confirm RPROMPT stop showing when the character in the user input are starting to get close to it.

![image](https://user-images.githubusercontent.com/45905567/204886697-cf71bf64-2f78-4d25-a1cd-515b6f95cea3.png)

**Comment by mirsella** [comment]
Thanks for the fast answer. Should I open a feature request ?

**Reply by JanDeDobbeleer** [reply]
@mirsella this is pretty challenging tbh. Especially to get it right on all shells for consistency. It would also require a continuation prompt for example. Not sure if this is something I'm eager to invest time in, allow me to do a short spike first.

**Reply by mirsella** [reply]
Of course, it was just an idea which I think may be useful, no worries if you think it's too complicated.
Maybe with `tput cols` which return the width of terminals on linux. I thought about a "max_cols" optional key.
We can get the output from tput cols and compare it with max_cols when starting to render the block : if max_cols != 0 && tput_cols > max_cols; then return and don't draw the block.

Like in the case below, to draw the username@hostname -> only a newline only if the terminal is smaller than 77 cols, which is on my pc, a terminal split in half
![image](https://user-images.githubusercontent.com/45905567/205034717-19e773d6-855e-4fe5-a1cd-3bfeac056dea.png)

**Comment by mirsella** [comment]
i made it work with 
```go
	if width, _, err := term.GetSize(int(os.Stdin.Fd())); err == nil && segment.MaxCols > 0 && width > segment.MaxCols {
		return
	}
```
and in the type of segment :
```go
	MaxCols             int            `json:"max_cols,omitempty"`
``` 
and a sample theme file: https://github.com/mirsella/dotfiles/blob/laptop/.config/zsh/lib/velvet.json with 77 cols set as max_cols
in the SetEnabled function in segment.go
it's perfect for me so i'll keep using my fork.
i can open a pull request if you want, but i don't know if it work on windows, it use https://pkg.go.dev/golang.org/x/term#section-readme.
first time coding in go btw i don't know if this is good code or not.

other way of getting the width of the terminal could be : 
https://github.com/gdamore/tcell
https://github.com/nathan-fiscaletti/consolesize-go
which are known to work on windows too.

**Reply by JanDeDobbeleer** [reply]
@mirsella open the PR, I'd love to have a look as I haven't got around to it yet.
